From c5496869eab66dbf2ee1cb6d7f10536da809ace0 Mon Sep 17 00:00:00 2001
From: syzygial <67280346+syzygial@users.noreply.github.com>
Date: Sat, 31 May 2025 22:38:52 -0400
Subject: [PATCH 4/9] Fix missing Lua::Lua target when using system Lua

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git CMakeLists.txt CMakeLists.txt
index 507d9be5..ed1a6b59 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -276,6 +276,10 @@ if (sol2-is-top-level-project)
 		set(LUA_LIBRARIES ${lualib})
 	endif()
 
+	if(NOT TARGET Lua::Lua)
+		add_library(Lua::Lua ALIAS ${lualib})
+	endif()
+
 	if (NOT LUA_FOUND AND NOT LUABUILD_FOUND)
 		message(FATAL_ERROR "sol2 Lua \"${SOL2_LUA_VERSION}\" not found and could not be targeted for building")
 	endif()
-- 
2.48.1

