From f433abdad78bf7601563674f292ac4c96b042220 Mon Sep 17 00:00:00 2001
From: syzygial <67280346+syzygial@users.noreply.github.com>
Date: Sat, 31 May 2025 12:04:26 -0400
Subject: [PATCH 5/9] fix INTERFACE_LINK_LIBRARIES property for lualib

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index ed1a6b59..1e2edf9a 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -270,7 +270,7 @@ if (sol2-is-top-level-project)
 		set_target_properties(${lualib}
 			PROPERTIES 
 			INTERFACE_INCLUDE_DIRECTORIES ${LUA_INCLUDE_DIR}
-			INTERFACE_LINK_LIBRARIES "${LUA_SEARCH_DEPENDENCY_LIBS} ${CMAKE_DL_LIBS}"
+			INTERFACE_LINK_LIBRARIES "${LUA_SEARCH_DEPENDENCY_LIBS};${CMAKE_DL_LIBS}"
 			IMPORTED_LINK_INTERFACE_LANGUAGES C
 			IMPORTED_LOCATION ${lualiblocation})
 		set(LUA_LIBRARIES ${lualib})
-- 
2.48.1

